src/main/java/com/artipie/RpPermissions.java

```diff
@@ -23,38 +23,70 @@
  */
 package com.artipie;
 
+import com.amihaiemil.eoyaml.Yaml;
+import com.amihaiemil.eoyaml.YamlMapping;
+import com.amihaiemil.eoyaml.YamlSequence;
 import com.artipie.http.auth.Permissions;
 import java.io.File;
-import org.apache.commons.lang3.NotImplementedException;
+import java.io.IOException;
 
 /**
  * Repository permissions: this implementation is based on
  * on repository yaml configuration file.
  * @since 0.2
- * @todo #69:30min Implement this interface to read and process permissions from
- *  repository yaml configuration file. Test is already implemented, see {@link RpPermissionsTest},
- *  don't forget to enable is when this class is ready. Remove also PMD suppressions please.
- *  For more details check issue #69.
  */
-@SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
 public final class RpPermissions implements Permissions {
 
     /**
-     * Repository yaml configuration file.
+     * YAML storage settings.
      */
-    private final File conf;
+    private final YamlMapping yaml;
 
     /**
      * Ctor.
-     * @param conf Conf file
+     * @param conf Config file
      */
     public RpPermissions(final File conf) {
-        this.conf = conf;
+        this(readYaml(conf));
```

Comment:
@Guseyn I've designed this object to work with yaml (I mean yaml as a java object), when accepting `File` in the ctor, I have to validate the file - it can be not even yaml, then there is no point instantiating this object. Validations in ctors are allowed, Yegor explained it [here](https://www.yegor256.com/2018/05/29/object-validation.html). 
And I do not want to move file parsing in `allowed` method, because there is no point to parse it each time we want to check permissions.